### PR TITLE
Feature/news: Space redactor

### PR DESCRIPTION
**Pull request by aelbenney**
Add the redactor role for spaces and hide/show composer if user is redactor or not.

### CODE REVIEW

File: component/service/src/main/java/org/exoplatform/social/rest/entity/ProfileEntity.java [hunk 1]
```diff
@@ -152,6 +152,15 @@ public Boolean getIsSpacesManager() {
     return (Boolean) getProperty("isSpacesManager");
   }
 
+  public ProfileEntity setIsSpaceRedactor(boolean isSpaceRedactor) {
+    setProperty("IsSpaceRedactor", isSpaceRedactor);
```
Comment (rdenarie) [inline component/service/src/main/java/org/exoplatform/social/rest/entity/ProfileEntity.java]:
```suggestion
    setProperty("isSpaceRedactor", isSpaceRedactor);
```

File: component/service/src/main/java/org/exoplatform/social/rest/entity/ProfileEntity.java [hunk 2]
```diff
@@ -152,6 +152,15 @@ public Boolean getIsSpacesManager() {
...
+    return this;
+  }
+
+  public Boolean getIsSpaceRedactor() {
+    return (Boolean) getProperty("IsSpaceRedactor");
```
Comment (rdenarie) [inline component/service/src/main/java/org/exoplatform/social/rest/entity/ProfileEntity.java]:
```suggestion
    return (Boolean) getProperty("isSpaceRedactor");
```

File: extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl [hunk 3]
```diff
@@ -24,11 +24,29 @@
...
+  import org.exoplatform.services.security.Identity;
+  import org.exoplatform.services.security.ConversationState;
 
   ExoFeatureService featureService = ExoContainerContext.getCurrentContainer().getComponentInstanceOfType(ExoFeatureService.class);
   boolean isMainStreamComposerEnabled = System.getProperty("exo.config.mainStreamComposer.enabled").equals("true");
+  //check if space has redactors
+  Boolean spaceHasRedactors = false;
+  Boolean isCurrentUserRedactor = false;
+  if(uicomponent.getSpaceGroupId()) {
+    String SpaceGroupId = uicomponent.getSpaceGroupId();
```
Comment (rdenarie) [inline extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl]:
```suggestion
    String spaceGroupId = uicomponent.getSpaceGroupId();
```

File: extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl [hunk 4]
```diff
@@ -24,11 +24,29 @@
...
+    List<String> spaceRedactors = SpaceUtils.findMembershipUsersByGroupAndTypes(SpaceGroupId, "redactor");
```
Comment (rdenarie) [inline extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl]:
```suggestion
    List<String> spaceRedactors = SpaceUtils.findMembershipUsersByGroupAndTypes(spaceGroupId, "redactor");
```

File: extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl [hunk 5]
```diff
@@ -24,11 +24,29 @@
...
+    if (!spaceRedactors.isEmpty()) {
+      spaceHasRedactors = true;
+      //check if user connected is redactor!!!!
+      ConversationState currentState = ConversationState.getCurrent();
+      String currentUserId = currentState.getIdentity().getUserId();
```
Comment (rdenarie) [inline extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl]:
```suggestion
      String currentUserId = _ctx.getRequestContext().getRemoteUser();
```

File: webapp/portlet/src/main/resources/locale/portlet/social/PeopleListApplication_en.properties [hunk 6]
```diff
@@ -11,6 +11,8 @@ peopleList.button.cancelRequest=Cancel request
 peopleList.button.showMore=Show more
 peopleList.button.promoteManager=Promote manager
 peopleList.button.removeManager=Remove manager
+peopleList.button.setAsRedactor=Set as redactor
+peopleList.button.removeRedactor=Remove redactor
```
Comment (rdenarie) [inline webapp/portlet/src/main/resources/locale/portlet/social/PeopleListApplication_en.properties:15]: Can you add it also in fr file ?

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: component/service/src/main/java/org/exoplatform/social/rest/entity/ProfileEntity.java (hunk 1 above)
Comment (aelbenney) [reply component/service/src/main/java/org/exoplatform/social/rest/entity/ProfileEntity.java]: done

File: component/service/src/main/java/org/exoplatform/social/rest/entity/ProfileEntity.java (hunk 2 above)
Comment (aelbenney) [reply component/service/src/main/java/org/exoplatform/social/rest/entity/ProfileEntity.java]: done

File: extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl (hunk 3 above)
Comment (aelbenney) [reply extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl]: done

File: extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl (hunk 4 above)
Comment (aelbenney) [reply extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl]: done

File: extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl (hunk 5 above)
Comment (aelbenney) [reply extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl]: done

**Review by rdenarie — APPROVED** [review]

**Pull request merged**